### PR TITLE
fix(wallet): Show Import Account Error

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Fixes failed account `Import` not showing `Error` and closing `Import Modal`

Resolves <https://github.com/brave/brave-browser/issues/24402>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Go to the `Accounts` tab, click `Import` and select `Solana`
2. Type in something random for the `Private Key` enter a `Account Name` and click `Import`
3. It should show an import `Error` and not close the `Import Modal`
4. Now paste in a real `Private Key`, it should import and the modal should close. 

Before:

https://user-images.githubusercontent.com/40611140/182495127-fa983566-3998-41d9-a3ae-9ca1aac4b940.mov

After:

https://user-images.githubusercontent.com/40611140/182502602-7a3e8029-86c6-4b02-a3b5-01c26f163bd1.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx [hunk 1]
```diff
@@ -190,6 +192,13 @@ export const ImportAccountModal = () => {
     history.push(WalletRoutes.ImportAccountModal.replace(':accountTypeName?', accountType.name.toLowerCase()))
   }, [])
 
+  React.useEffect(() => {
+    if (hasImportError === false) {
+      history.push(WalletRoutes.Accounts)
+      setImportError(undefined)
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx:198]: shouldn't we dispatch before navigating?

File: components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-hardware-account-modal.tsx [hunk 2]
```diff
@@ -57,12 +62,12 @@ export const AddHardwareAccountModal = () => {
...
 
   // methods
-  const setImportError = React.useCallback((hasError: boolean) => {
+  const setImportError = React.useCallback((hasError: ImportAccountErrorType) => {
     dispatch(WalletPageActions.setImportAccountError(hasError))
   }, [])
 
   const closeModal = React.useCallback(() => {
-    setImportError(false)
+    setImportError(undefined)
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-hardware-account-modal.tsx:70]: setImportError is missing from dep array.

File: components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx [hunk 3]
```diff
@@ -92,12 +99,12 @@ export const ImportAccountModal = () => {
...
 
   // methods
-  const setImportError = React.useCallback((hasError: boolean) => {
+  const setImportError = React.useCallback((hasError: ImportAccountErrorType) => {
     dispatch(WalletPageActions.setImportAccountError(hasError))
   }, [])
 
   const onClickClose = React.useCallback(() => {
-    setImportError(false)
+    setImportError(undefined)
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx:107]: setImportError missing from deps array

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx]: Good call, switched!

File: components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-hardware-account-modal.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-hardware-account-modal.tsx:70]: Added!

File: components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx:107]: Added!

**Review by josheleonard — APPROVED** [review]

**Pull request merged**